woocommerce/woocommerce-admin
# Remove SlotFill need for clientside id's and make identifiers more unique

Fixes https://github.com/woocommerce/woocommerce-admin/issues/5626
Fixes https://github.com/woocommerce/woocommerce-admin/issues/5458

This is an alternate to https://github.com/woocommerce/woocommerce-admin/pull/5642

id's were added in wc-admin registered pages so that there could be identical ids matched between `wcNavigation.menuItems` (which is populated by PHP) and client functions `getPages` and `getReports`. Instead, this PR removes those id's and uses the `path` to determine uniqueness. This removes the requirement for wc-admin connected extensions to have to update their filters to add id's.

id's in PHP were also made less generic when being registered by the Nav. This avoids duplication, such as was the case with `Settings > Products`.

### Detailed test instructions:

1. Ensure all wc-admin pages can navigate without refreshing the page, including Home, Settings > Analytics, and Customers.
2. Ensure Settings > Products is a functioning Nav item.
